packages/rum-core/src/performance-monitoring/span-base.js

```diff
@@ -29,7 +29,16 @@ import { NAME_UNKNOWN, TYPE_CUSTOM } from '../common/constants'
 class SpanBase {
   // context
 
-  constructor(name = NAME_UNKNOWN, type = TYPE_CUSTOM, options = {}) {
+  constructor(name, type, options = {}) {
```

Comment:
default assignments checks for undefined values. Wont work for empty strings. Since our config-service has default values of `''` this check is moved here to handle all spans and transactions.